---
title: "Differential_geneexpression_sclerosis_samples"
author: "Nischal"
date: "2024-02-22"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
setwd("/Users/nischal/Downloads/Bioinformatics/Multiple_Sclerosis_human/results")
normal_sample <- read.delim("./normal_MS_gene_count_matrix.txt", header = TRUE)
```

```{r}
colnames(normal_sample)[2] <- "normal_MS"
head(normal_sample)

```

```{r}
setwd("/Users/nischal/Downloads/Bioinformatics/Multiple_Sclerosis_human/results")
progressive_sclerosis <- read.table("Multiple_sclerosis_gene_count_matrix.txt", header = TRUE, row.names = NULL, sep = "\t")
```

```{r}
colnames(progressive_sclerosis)[2] <- "progressive_MS"
head(progressive_sclerosis)
```
```{r}
# reading other genecount reads from other patients.
setwd("/Users/nischal/Downloads/MS/results")
patient4_7_reads <- read.delim("features4_7_count.tsv", header = TRUE)
head(patient4_7_reads)
```
merging all datasets

```{r}
df_list <- list(normal_sample, progressive_sclerosis, patient4_7_reads)
normalvsprogressiveMS <- Reduce(function(x, y) merge(x, y, all=TRUE), df_list)
```

# save file

saving the file

```{r}

write.csv(normalvsprogressiveMS, "/Users/nischal/Downloads/Bioinformatics/Multiple_Sclerosis_human/results/normal_progressive_MS_gene_count_5patients.tsv", row.names = FALSE)
```

```{r}
normalvsprogressiveMS <- read.csv( "/Users/nischal/Downloads/Bioinformatics/Multiple_Sclerosis_human/results/normal_progressive_MS_gene_count_5patients.tsv", header = TRUE)
```
performing the differential gene expression

```{r}
library(DESeq2)
library(ggplot2)
```

```{r}
rownames(normalvsprogressiveMS) <- normalvsprogressiveMS$Geneid

normalvsprogressiveMS <- normalvsprogressiveMS[, -1]
```
```{r}
normalvsprogressiveMS <- normalvsprogressiveMS[which(rowSums(normalvsprogressiveMS) >50), ]
head(normalvsprogressiveMS)
```


```{r}
colnames(normalvsprogressiveMS)
```
```{r}
condition <- factor(c("normal_MS", "progressive_Ms", "progressive_Ms", "progressive_Ms", "normal_MS", "progressive_Ms", "normal_MS", "progressive_Ms", "normal_MS", "normal_MS"))
```

  
```{r}
coldata <- data.frame(row.names = colnames(normalvsprogressiveMS), condition)
```

```{r}
coldata
```

creating dseq matrix

```{r}
dds <- DESeqDataSetFromMatrix(countData = normalvsprogressiveMS, colData = coldata, design = ~condition)
```

```{r}
dds <- DESeq(dds)
```

```{r}
vsdata <- vst(dds, blind = FALSE)
```

Quality Control

1. Plot PCA

```{r}
pca_plotted <- plotPCA(vsdata, intgroup = "condition")
```

doesn't see to separe much. not the ideal separation.

disperson plot
```{r}
plotDispEsts(dds)
```


```{r}
# diff expressed genes

diff_expressed_genes <- results(dds, contrast = c("condition", "progressive_Ms", "normal_MS"))
```

```{r}
diff_expressed_genes
```

```{r}
# filtering the signficiant genes

sig_genes <- na.omit(diff_expressed_genes)

```

```{r}
sig_genes <- sig_genes[sig_genes$padj < 0.05,]
sig_genes
```

no significantly differntially expressed genes???

```{r}
diff_expressed_genes <- na.omit(diff_expressed_genes)
diff_expressed_genes[diff_expressed_genes$padj < 0.05,]
```

```{r}
diff_expressed_genes[diff_expressed_genes$padj < 0.1,]
```


```{r}
diff_expressed_genes_p0.4 <- diff_expressed_genes[diff_expressed_genes$padj < 0.4,]
```

```{r}
write.csv(diff_expressed_genes_p0.4, "/Users/nischal/Downloads/Multiple_Sclerosis_human/results/not_so_diff_exp_genes_p0.4", row.names = TRUE)
```